### ISSUE TITLE
How can I fix this? Assertion `(vid == v) && (dir == d) && (pid == p)' failed

### ISSUE THREAD

**Issue by kjhkim0702** [SJTU-IPADS/wukong#2]
I get the following assertion failure: "wukong: /home/jhkim/wukong/core/gstore.hpp:107: ikey_t::ikey_t(sid_t, sid_t, dir_t): Assertion `(vid == v) && (dir == d) && (pid == p)' failed."
I founded that at some moment, pid != p; pid = 86069 and p=1134645. Is it related to NBITS_IDX in core/gstore.hpp and datagen/generate_data.cpp?
I used single machine configuration, and inferenced lubm80 dataset

**Comment by realstolz** [comment]
I guess the assertion at `wukong/core/gstore.hpp:101`, which is used to ensure that direction (dir, 1-bit), predicate ID (pid, 17-bit), and vertex ID (vid, 46-bit) are not **truncated** when building the key. However, the predicate ID should never be truncated since we also reserve the first 17 bits of ID for the predicate/type string when generating ID-format data (`NBITS_IDX=17` in `datagen/generte_data.cpp`).

The predicate ID (1134645) is clearly too large for lubm80. In fact, the LUBM dataset only requires 32 predicate ID. You could first check the file named "str_index", which includes all mapping from predicate/type strings to predicate IDs. 

Could you check the file `str_index` and ensure the data generation is correct. For example, the output of `./generate_data`. Each `id_uniXX.nt` file contains the RDF triples of one university. The 2nd column indicates the predicate ID, which should be very small (<32).

Please also provide your commands to generate data and run Wukong.

**Comment by kjhkim0702** [comment]
Thank you for the comment and sorry for late response. The reason for the failure was the dataset. I did not generate lubm80 dataset, but I used a dataset that was created by other person. The dataset contained some triples that caused wrong id mappings. I tried to find the wrong triples by my own and it took longer time to find them than I expected; and I forgot to leave response.. Sorry for the late response again.
